### PR TITLE
ci: adds release-please workflow

### PR DESCRIPTION
**Pull request by ctran88**
## What's New?

Adds a new `release` workflow that uses [release-please](https://github.com/googleapis/release-please) to manage the versioning, changelogs, and publication

It will currently only be manually triggered since our upstream repo is still configured to trigger the `deploy` workflow.

I will switch over the trigger once all of the SDKs are good to go with the release workflow.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have manually tested my code thoroughly
- [ ] I have added/updated inline documentation for public facing interfaces if relevant
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing integration and unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -92,7 +84,6 @@
         "supertest": "^7.0.0",
         "ts-jest": "^29.2.5",
         "ts-node-dev": "^2.0.0",
-        "tsup": "^8.2.4",
```
Comment (flanagankp) [inline package.json:95]: tsup is still being used for the build script (and I have no plan to change this)

Comment (ctran88) [reply package.json]: ah right you are. i missed the parallel tsup build in the script. fixed in [9655efe](https://github.com/passageidentity/passage-node/pull/182/commits/9655efef48713c422f0acf36674db74f27c255ef)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+name: Release
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
```
Comment (flanagankp) [inline .github/workflows/release.yml:7]: why is this set to run on pushes to main? We don't want to release on every push to main, right?

Comment (flanagankp) [reply .github/workflows/release.yml:7]: Thanks for the explanation, sounds good to me

**Review by flanagankp — APPROVED** [review]

**Comment by ctran88** [comment]
added beachball back in so the `deploy` workflow will continue to work until we deprecate it

**Pull request merged**